docker/docker_client.go

```diff
@@ -258,6 +271,41 @@ func newDockerClient(ctx *types.SystemContext, ref dockerReference, write bool,
 	}, nil
 }
 
+// CheckAuth validates credentials
+func CheckAuth(ctx context.Context, sCtx *types.SystemContext, username, password, registry string) error {
+	newLoginClient, err := newDockerClient(sCtx, dockerReference{}, false, username, password, registry, "login")
+	if err != nil {
+		return errors.Wrapf(err, "error creating new docker client")
+	}
+
+	resp, err := newLoginClient.makeRequest(ctx, "GET", "/v2/", nil, nil)
```

Comment:
`makeRequest` calls `detectProperties` which makes a request to `/v2/`, so this seems to do the same thing twice.